elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/core/QueryLogger.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2020, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.datastores.aggregation.core;
+
+import com.yahoo.elide.datastores.aggregation.query.Query;
+
+import java.security.Principal;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import javax.ws.rs.core.MultivaluedMap;
+
+/**
+ * Query Logger Interface for Elide
+ */
+public interface QueryLogger {
+
+    /**
+     * Accepts the incoming GraphQL query and notes the start time for the query
+     * @param queryId The RequestScope requestId.
+     * @param user The Principal user
+     * @param headers Http Request Headers
+     * @param apiVer API Version
+     * @param apiQuery QueryString of the requested URL request
+     */
+    void acceptQuery(UUID queryId, Principal user, Map<String, String> headers,
+                     String apiVer, String apiQuery);
+
+    /**
+     * Accepts the incoming JSON API query and notes the start time for the query
+     * @param queryId The RequestScope requestId.
+     * @param user The Principal user
+     * @param headers Http Request Headers
+     * @param apiVer API Version
+     * @param queryParams QueryParams for the incoming JSON API query
+     * @param path The apiQuery endpoint path for the incoming query
+     */
+    void acceptQuery(UUID queryId, Principal user, Map<String, String> headers, String apiVer,
```

Comment:
I'm wondering if it would be more powerful to log the elide User object here instead of the Principal.  The user has some additional bits (like the role support).